### PR TITLE
[WIP] schema2* tool update: include schema aliases and anonymous schemas. 

### PR DESCRIPTION
**Pull request by alxrsngrtn**
A first iteration towards a solution for issue #3663.

My plan is to introduce the solution in a few steps: 
First, collect a minimal set of schemas for generation in a single collection (achieved in this PR). 
Second, parse the collection of target schemas, and produce a global strategy for classes (and interfaces) to generate. 
Third, generate the structures in each target language.

### CODE REVIEW

File: src/tools/schema2base.ts [hunk 1]
```diff
@@ -38,51 +41,110 @@ export abstract class Schema2Base {
...
+   * Includes schemas that have no names (anonymous) or schemas with multiple aliases
+   *
+   * @param manifest Manifest expended by loader.
+   * @return Dictionary<Schema> target schemas for code generation.
+   */
+  private static collectSchemas(manifest: Manifest): Dictionary<Schema> {
+    /** Helper function: produce a new name for a schema found in the manifest. */
+    const mangleDuplicateName = <T>(collection: Dictionary<T>, name: string): string => {
+      let candidate = name;
+      while (candidate in collection) candidate += '_';
```
Comment (piotrswigon) [inline src/tools/schema2base.ts]: I don't understand why we have to do such things. Let's VC?

Comment (alxrsngrtn) [reply src/tools/schema2base.ts]: Sure, let's VC.

**Comment by alxrsngrtn** [comment]
Closing this PR to clean up the git history. Will apply the feedback changes to the new branch: `schema-gen`.

File: src/tools/schema2base.ts (hunk 1 above)
Comment (alxrsngrtn) [reply src/tools/schema2base.ts]: For git historians out there, the takeaways from conversations with Myk and Piotr were: 
- need to construct some sort of data structure to mirror the type lattice within particles
- type compatibility is only needed within particles. A single particles input and output types should agree, given their definition. But we need not unify types across particles. Why? Runtime takes care of that. 
- best follow the example in the original [GH issue](https://github.com/PolymerLabs/arcs/issues/3663#issuecomment-538465035) for guidance

**Pull request closed without merging**